soak/bootstrap.sh

```diff
@@ -0,0 +1,213 @@
+    helm repo add $PROM_REPO_NAME $PROM_REPO_URL 1> /dev/null
+    echo "ok."
+fi
+
+# Install the prometheus chart
+ret=0
+helm list -n $PROM_NAMESPACE 2> /dev/null | grep -q $PROM_CHART_RELEASE_NAME || ret=$?
+if [ $ret -eq 0 ]; then
+    echo "Prometheus Helm release ($PROM_CHART_RELEASE_NAME) already installed in cluster"
+else
+    echo -n "Installing Prometheus chart ... "
+    helm install --create-namespace -n $PROM_NAMESPACE \
+        --set prometheus.prometheusSpec.additionalScrapeConfigs[0].job_name="ack-controller" \
+        --set prometheus.prometheusSpec.additionalScrapeConfigs[0].static_configs[0].targets[0]="$AWS_SERVICE-controller-metrics.ack-system:8080" \
+        $PROM_CHART_RELEASE_NAME prometheus-community/kube-prometheus-stack 1> /dev/null
+    echo "ok."
+fi
+
+# Apply the grafana dashboard
+kubectl apply -n $PROM_NAMESPACE -k github.com/aws-controllers-k8s/test-infra/soak/monitoring/grafana?ref=main
+
+# Build and publish the soak test runner image
+SOAK_IMAGE_REPO_URI="$($AWS_ECR_PUBLIC_CLI describe-repositories --repository-name $SOAK_IMAGE_REPO_NAME --output text --query "repositories[0].repositoryUri")" || \
+    { >&2 echo "Could not get the soak test image repository URI"; exit 1; }
+
+echo "Building soak test image ... "
+$OCI_BUILDER build --platform $SOAK_IMAGE_PLATFORM -t $SOAK_IMAGE_REPO_URI:$SOAK_IMAGE_TAG \
+    --build-arg AWS_SERVICE=$AWS_SERVICE --build-arg E2E_GIT_REF=$CONTROLLER_TAG "$TEST_INFRA_DIR/soak"
+echo "ok."
+
+echo -n "Pushing soak test image to ECR public ... "
+$AWS_ECR_PUBLIC_CLI get-login-password | $OCI_BUILDER login --username AWS --password-stdin public.ecr.aws 1> /dev/null 2>&1
+$OCI_BUILDER push $SOAK_IMAGE_REPO_URI:$SOAK_IMAGE_TAG 1> /dev/null
+echo "ok."
+
+# Install the soak test runner
+ret=0
+helm list -n $CONTROLLER_NAMESPACE 2> /dev/null | grep -q $SOAK_RUNNER_CHART_RELEASE_NAME || ret=$?
+if [ $ret -eq 0 ]; then
+    echo -n "Controller Helm release ($SOAK_RUNNER_CHART_RELEASE_NAME) already installed in cluster"
+else
+    echo -n "Installing soak test chart ... "
+    helm upgrade --install --create-namespace -n $CONTROLLER_NAMESPACE \
+        --set awsService=$AWS_SERVICE --set soak.imageRepo=$SOAK_IMAGE_REPO_URI \
+        --set soak.imageTag=$SOAK_IMAGE_TAG --set soak.startTimeEpochSeconds=$(date +%s) \
+        --set soak.durationMinutes=$NET_SOAK_TEST_DURATION_MINUTES \
+        $SOAK_RUNNER_CHART_RELEASE_NAME "$SOAK_DIR/helm/ack-soak-test" 1> /dev/null
+    echo "ok."
+fi
+
+# Final messages and links
+echo "Your soak test cluster should now be operational and actively running tests"
+echo "To port-forward your Grafana dashboard use the following command:"
+echo "    kubectl port-forward -n $PROM_NAMESPACE service/$PROM_CHART_RELEASE_NAME-grafana $LOCAL_GRAFANA_PORT:80 --address='0.0.0.0' >/dev/null &"
+echo "Then navigate to http://localhost:$LOCAL_GRAFANA_PORT/"
+echo ""
+echo "The default username is \"admin\" and the default password is \"prom-operator\""
```

Comment:
I believe a `soak/cleanup.sh` script is needed as well.

Reply:
Apart from deleting the cluster, what else would that achieve? I imagine we want to keep the cluster running, for future soak tests?

Reply:
* uninstall soak-test-runner chart
* uninstall the controller-chart
* cleanup the port-forward bg processes if any.

above 3 will allow the cluster to be reused without old installations coming in the way.

Reply:
What about I just focus on writing the script in such a way that we can re-use it for each release? Upgrading charts where applicable?

Also port-forwarding is not covered inside the script because ideally we would set up an ingress and put it behind a load balancer and domain name, so that it can be accessed by anyone. But that's for future us